interpreter/cling/lib/Interpreter/IncrementalJIT.cpp

```diff
@@ -577,6 +581,18 @@ IncrementalJIT::IncrementalJIT(
 
   char LinkerPrefix = this->m_TM->createDataLayout().getGlobalPrefix();
 
+#if defined(__linux__) && defined(__GLIBC__)
+  // Inject lstat to pass test/DynamicLibraryManager/cached_realpath.C, it may
+  // be in libc_nonshared.a where symbols are not found automatically. See also
+  // upstream issue https://github.com/llvm/llvm-project/issues/61289.
+  SymbolMap LibcNonsharedSymbols;
+  LibcNonsharedSymbols[Jit->mangleAndIntern("lstat")] = JITEvaluatedSymbol(
+      static_cast<JITTargetAddress>(reinterpret_cast<uintptr_t>(&lstat)),
+      JITSymbolFlags::Exported);
+  cantFail(
+      Jit->getMainJITDylib().define(absoluteSymbols(LibcNonsharedSymbols)));
```

Comment:
I wonder whether we could reuse `IncrementalJIT::addOrReplaceDefinition()` for this, but other than that LGTM!